From 58b76e8fd5359e58f0807d935ca14a3d0d7aed82 Mon Sep 17 00:00:00 2001
From: Andrew Butt <andrewb1999@gmail.com>
Date: Tue, 23 Feb 2021 16:19:55 -0500
Subject: [PATCH] rvpld patch

---
 litex/soc/interconnect/axi.py   | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/litex/soc/interconnect/axi.py b/litex/soc/interconnect/axi.py
index 4847c8d5..450992dc 100644
--- a/litex/soc/interconnect/axi.py
+++ b/litex/soc/interconnect/axi.py
@@ -109,6 +109,13 @@ def connect_to_pads(bus, pads, mode="master", axi_full=False):
                 r.append(sig.eq(pad))
             else:
                 r.append(pad.eq(sig))
+        for name, width in [("valid", 1)]:
+            sig  = getattr(ch, name)
+            pad  = getattr(pads, channel + name)
+            if mode == "master":
+                r.append(pad.eq(sig))
+            else:
+                r.append(sig.eq(pad))
     return r
 
 def _axi_layout_flat(axi):
-- 
2.30.1

